### ISSUE TITLE
Please add number of years to anniversaries

### ISSUE THREAD

**Issue by ivanslamek** [plusonelabs/calendar-widget#227]
Hi.

Can you please add number of years (e.g. in brackets beside the event) for anniversary events (such as birthdays etc.) so you can see it right away (which anniversary it is)?

Have a nice day.

**Comment by adamdehaven** [comment]
Just update the event text in your calendar

**Comment by ivanslamek** [comment]
What do you mean? Should I update text for recurring events (anniversaries) every year? I hope not...

**Comment by adamdehaven** [comment]
I'm not sure how this would be done. Google Calendar doesn't even support this feature on it's own

**Comment by ivanslamek** [comment]
All my anniversaries come from contacts in my phone. I had one calendar widget which displayed the anniversary number (I think it was Agenda widget or similar).

**Comment by yvolk** [comment]
Joining this with #114 as showing age for Birthdays and Anniversaries looks like very close tasks

**Comment by virtualdj** [comment]
> Joining this with #79 as showing age for Birthdays and Anniversaries looks like very close tasks

I don't think it's the same, or better... birthdays are stored in Google Calendar and **tied with contacts**. But one (like me) could have a shared calendar with **yearly recurring events** and if the widget is able to count the "instance number" of these recursions and display it in square brackets it would be very useful!

You could for example use it to count any age, of your pet, house, anything... without depending on a phone contact.

Please, change your mind about closing this issue.

> I had one calendar widget which displayed the anniversary number (I think it was Agenda widget or similar).

Yes, it's **Pure Calendar Widget**. See my [question here](https://softwarerecs.stackexchange.com/q/50324): it would be great to have a feature like this in Calendar-Widget, too.

**Comment by yvolk** [comment]
@virtualdj As our widget cannot know a difference between "Anniversary" and any other recurring event, your suggestion better call:

Add an option: "Show occurrence number of a recurring event"

?!

I found good example cases here:
1. https://answers.microsoft.com/en-us/msoffice/forum/all/is-there-any-way-to-show-the-occurrence-number-of/47618036-12e2-4286-9826-444ee68aa4c6?auth=1

2. https://webapps.stackexchange.com/questions/10850/increment-number-inside-a-recurring-event-in-google-calendar

**Comment by virtualdj** [comment]
> your suggestion better call:
> Add an option: "Show occurrence number of a recurring event"

Yes, of course, but this also answers the OP requests because it can be useful for anniversaries, birthdays or any yearly event.

>     2. https://webapps.stackexchange.com/questions/10850/increment-number-inside-a-recurring-event-in-google-calendar

This is about changing the displayed text into Google Calendar (i.e. the event name), while I'm asking only to **change the display on the widget**.
So it's not something done at the server side (Google Calendar) but on the client (your widget): it should count the number of repetitions of a recurring event and display it (if the user enabled the relevant option).

>     1. https://answers.microsoft.com/en-us/msoffice/forum/all/is-there-any-way-to-show-the-occurrence-number-of/47618036-12e2-4286-9826-444ee68aa4c6?auth=1

Yes, something like that, for Android.

I don't know exactly how to implement the counter, probably like [THIS](https://stackoverflow.com/a/45111578)? As I said, Pure Calendar already does it so there has to be a way.
